File: 0566-reshape-the-matrix/0566-reshape-the-matrix.py
class Solution:
    def matrixReshape(self, mat: List[List[int]], r: int, c: int) -> List[List[int]]:
        if len(mat) * len(mat[0]) != r * c : return mat
        res = [[0 for j in range(c)] for i in range(r)]
        m = 0
        n = 0
        for i in range(len(mat)) :
            for j in range(len(mat[0])) :
                if n == c :
                    m += 1
                    n = 0
                res[m][n] = mat[i][j]
                n += 1
        return res

File: 0566-reshape-the-matrix/0566-reshape-the-matrix.cpp
class Solution {
public:
    vector<vector<int>> matrixReshape(vector<vector<int>>& mat, int r, int c) {
        if((mat.size() * mat[0].size()) != r * c) return mat;
        vector<vector<int>> res(r, vector<int>(c));
        int m = 0, n = 0;
        for(int i=0; i<mat.size(); i++) {
            for(int j=0; j<mat[0].size(); j++) {
                if(n == c) {
                    m += 1;
                    n = 0;
                }
                res[m][n++] = mat[i][j];
            }
        }
        return res;
    }
};